include/deal.II/base/quadrature_lib.h

```diff
@@ -511,6 +511,115 @@ class QTelles: public Quadrature<dim>
 
 /*@}*/
 
+/**
+* Gauss-Chebyshev quadrature rules integrate the weighted product
+* $\int_{-1}^1 f(x) w(x) dx$ with weight given by:
+* $w(x) = 1/\sqrt{1-x^2}$. The nodes and weights are known analytically,
+* and are exact for monomials up to the order $2n-1$, where $n$ is the number
+* of quadrature points.
+* Here we rescale the quadrature formula so that it is defined on
+* the interval $[0,1]$ instead of $[-1,1]$. So the quadrature formulas
+* integrate exactly the integral $\int_0^1 f(x) w(x) dx$ with the weight:
+* $w(x) = 1/sqrt{x(1-x)}$.
+* For details see:
+* M. Abramowitz & I.A. Stegun: Handbook of Mathematical Functions, par. 25.4.38
+*
+* @author Giuseppe Pitton, Luca Heltai 2015
+**/
+template <int dim>
+class QGaussChebyshev : public Quadrature<dim>
+{
+public:
+  /// Generate a formula with <tt>n</tt> quadrature points
+  QGaussChebyshev(const unsigned int n);
+
+private:
+  /// Sets the points of the quadrature formula.
```

Comment:
The comment here and elsewhere is now wrong. The function doesn't set anything -- it just "returns" something.